src/extremeweatherbench/utils.py

```diff
@@ -578,10 +579,204 @@ def maybe_convert_to_path(value: str | Path) -> str | Path:
     return value
 
 
-# Type alias for use in type hints
-PathOrStr = str | Path
+def _default_preprocess(input_data: IncomingDataInput) -> IncomingDataInput:
+    """Default forecast preprocess function that does nothing."""
+    return input_data
 
 
-def _default_preprocess(ds: xr.Dataset) -> xr.Dataset:
-    """Default forecast preprocess function that does nothing."""
-    return ds
+def _filter_kwargs_for_callable(kwargs: dict, callable_obj: Callable) -> dict:
+    """Filter kwargs to only include arguments that the callable can accept.
+
+    This method uses introspection to determine which arguments the callable
+    can accept and filters kwargs accordingly.
+
+    Args:
+        kwargs: The full kwargs dictionary to filter
+        callable_obj: The callable (function, method, etc.) to check against
+
+    Returns:
+        A filtered dictionary containing only the kwargs that the callable can accept
+    """
```

Comment:
A private method with a doc string is a bit of a code smell. Should probably be public.